pkg/providers/subnet/suite_test.go

```diff
@@ -116,63 +117,139 @@ var _ = Describe("Subnet Provider", func() {
 		subnets, err := awsEnv.SubnetProvider.List(ctx, nodeTemplate)
 
 		Expect(err).To(BeNil())
-		Expect(subnets).To(HaveLen(1))
-		Expect(aws.StringValue(subnets[0].SubnetId)).To(Equal("subnet-test1"))
-		Expect(aws.StringValue(subnets[0].AvailabilityZone)).To(Equal("test-zone-1a"))
+		Expect(subnets).To(Equal([]*ec2.Subnet{
```

Comment:
For these tests, I'm less convinced that we need to ensure the entire API surface. I had the opinion that we should ensure the entire API surface for the `nodeTemplate.status` because:
1. We control the API, so we should ensure the API
2. It's a smaller API surface, so it's easier to ensure every field

For these test, I think it might be sufficient to ensure the pieces that we care about and rely on. If we rely on all these fields in other places, then this is fine, we can keep it as you have it. But, if I recall, we really only care about the AZs and the IDs for these tests.

Reply:
Ah I see, we only really rely on  `SubnetId`, `AvailabilityZone`, and `AvailableIpAddressCount` for choosing the subnet, so I'll just add that pieces.